src/plugins/controls/public/react_controls/control_group/get_control_group_factory.tsx

```diff
@@ -68,19 +75,19 @@ export const getControlGroupEmbeddableFactory = (services: {
       } = initialRuntimeState;
 
       const autoApplySelections$ = new BehaviorSubject<boolean>(autoApplySelections);
-      const parentDataViewId = apiPublishesDataViews(parentApi)
-        ? parentApi.dataViews.value?.[0]?.id
-        : undefined;
-      const controlsManager = initControlsManager(
-        initialChildControlState,
-        parentDataViewId ?? (await services.dataViews.getDefaultId())
+      const defaultDataViewId = await services.dataViews.getDefaultId();
+      const lastSavedControlsState$ = new BehaviorSubject<ControlPanelsState>(
+        initialChildControlState
```

Comment:
Should this be `lastSavedRuntimeState.initialChildControlState` rather than `initialRuntimeState.initialChildControlState`? Because we are currently seeing the following weirdness when navigating away with unsaved changes and returning to the same dashboard:

https://github.com/user-attachments/assets/e82f7006-68ac-40dc-8bc2-bf864da55cfb

Reply:
On a related note, even after switching to `lastSavedRuntimeState`, the changes do get reset as expected (unlike in the attached video) but the control group is still reporting unsaved changes 🤔

Reply:
Actually, no.... this is definitely more complicated than that. Ahh! BRB while I process 🙈

Reply:
Ahh, I think I was actually on the right track - I think this **should** be `lastSavedRuntimeState.initialChildControlState`, but then `lastSavedRuntimeState` is no longer our source of truth in `initControlsManager` - for that, we still need `initialRuntimeState.initialChildControlState`. Does that make sense? 😅

Reply:
resolved in https://github.com/elastic/kibana/pull/190273/commits/8c4265cbdb45add439e79dae58791abad2844db0

I added a unit test to test "should remove previous sessions's unsaved changes on reset" to capture this complex case.